pulsar-broker/src/main/java/org/apache/pulsar/broker/loadbalance/impl/SimpleResourceAllocationPolicies.java

```diff
@@ -53,6 +54,20 @@ private Optional<NamespaceIsolationPolicies> getIsolationPolicies(String cluster
         }
     }
 
+    private CompletableFuture<Optional<NamespaceIsolationPolicies>> getIsolationPoliciesAsync(String clusterName) {
+        return this.pulsar.getPulsarResources().getNamespaceResources()
+                .getIsolationPolicies().getIsolationDataPoliciesAsync(clusterName);
```

Comment:
I'm not sure the callback of this CompletableFuture is still execute on the metadataStore thread. if yes, if switch callback execute thread to other thread is needed ? WDYT？